### PR TITLE
Fix game mode message on connect. Fix game mode command. 

### PR DESCRIPTION
**Pull request by MWisBest**
On connect, a message was sent to the player saying their game mode was changed. This fixes that.

Also, the game mode command did not work with simply "/gamemode creative". This fixes that as well.

### CODE REVIEW

File: src/main/java/org/spout/vanilla/command/AdministrationCommands.java [hunk 1]
```diff
@@ -255,7 +255,7 @@ public void gamemode(CommandContext args, CommandSource source) throws CommandEx
 				throw new CommandException(args.getString(0) + " is not online.");
 			}
 		} else {
-			if (!(source instanceof VanillaPlayer)) {
+			if (!(source instanceof Player)) {
```
Comment (ZNickq) [inline src/main/java/org/spout/vanilla/command/AdministrationCommands.java:258]: This should probably be VanillaPlayer, and you could change the message below to "you must be a vanilla player ..." :)
Other than that, looks fine!

Comment (windy1) [inline src/main/java/org/spout/vanilla/command/AdministrationCommands.java:258]: No, this was a mistake on my part. It is right to be player. ;)

EDIT: To clarify, Player will never be a VanillaPlayer but rather player.getEntity().getController() will be.

Comment (ZNickq) [inline src/main/java/org/spout/vanilla/command/AdministrationCommands.java:258]: Hmm? how? O: Isn't player the SpoutAPI player?
LE: oh, i failed -.- 
Hmm, get on irc, lawl

Comment (windy1) [inline src/main/java/org/spout/vanilla/command/AdministrationCommands.java:258]: Think of player as an added level of abstraction basically. `SpoutPlayer implements Player` -> `Player.getEntity()` (Player HAS-AN Entity) -> `Entity.getController()` (Entity HAS-A Controller) (Would be a PlayerController in this case)

An entity can have a PlayerController as their controller (ie. VanillaPlayer) but Player is not actually a controller itself.

**Pull request merged**